server/src/main/java/org/elasticsearch/index/mapper/CompositeSyntheticFieldLoader.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.apache.lucene.index.LeafReader;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static java.util.Collections.emptyList;
+
+/**
+ * A {@link SourceLoader.SyntheticFieldLoader} that uses a set of sub-loaders
+ * to produce synthetic source for the field.
+ * Typical use case is to gather field values from doc_values and append malformed values
+ * stored in a different field in case of ignore_malformed being enabled.
+ */
+public class CompositeSyntheticFieldLoader implements SourceLoader.SyntheticFieldLoader {
```

Comment:
I noticed after implementing this that this is very close to what `ObjectMapper.SyntheticSourceFieldLoader` does. Maybe we can unify some code later.

This is also an alternative approach to current implementation of f.e. `SortedNumericDocValuesSyntheticFieldLoader` where malformed values handling is implemented explicitly. That logic is repeated in multiple loaders that handle different doc values types. I obviously didn't refactor that in this PR but wanted to gather some thoughts.

Reply:
+1 I think in a followup we can explore how to have a common base class for this class and `ObjectMapper.SyntheticSourceFieldLoader`.